identity-iota/src/did/doc/iota_document.rs

```diff
@@ -67,38 +67,30 @@ impl TryMethod for IotaDocument {
 }
 
 impl IotaDocument {
-  /// Creates a new DID Document from the given KeyPair.
-  ///
-  /// The DID Document will be pre-populated with a single authentication
-  /// method based on the provided [KeyPair].
-  ///
-  /// The authentication method will have the DID URL fragment `#authentication`
-  /// and can be easily retrieved with [Document::authentication].
-  pub fn from_keypair(keypair: &KeyPair) -> Result<Self> {
-    let method: IotaVerificationMethod = IotaVerificationMethod::from_keypair(keypair, "authentication")?;
-
-    // SAFETY: We don't create invalid Methods.  Method::from_keypair() uses the MethodBuilder
-    // internally which verifies correctness on construction.
-    Ok(unsafe { Self::from_authentication_unchecked(method) })
-  }
+  const DEFAULT_METHOD_FRAGMENT: &'static str = "authentication";
 
-  /// Creates a new DID Document from the given KeyPair and network.
+  /// Creates a new DID Document from the given [`KeyPair`] and network, defaulting to
+  /// [`Network::Mainnet`](crate::tangle::Network::Mainnet) if unspecified.
   ///
-  /// The DID Document will be pre-populated with a single authentication
-  /// method based on the provided [KeyPair].
+  /// The DID Document will be pre-populated with a single authentication verification method
+  /// derived from the provided [`KeyPair`]. This method will have the DID URL fragment
+  /// `#authentication` and can be easily retrieved with [`Document::authentication`].
```

Comment:
This doc comment will be outdated after #421 is merged. So something along the lines of this would be better:

```suggestion
  /// The DID Document will be pre-populated with a single verification method
  /// derived from the provided [`KeyPair`], with an attached authentication relationship.
  /// This method will have the DID URL fragment `#authentication` and can be easily
  /// retrieved with [`Document::authentication`].
```